File: fixtures/scale/julia/large/src/core/Csv.jl
module BenchCsv

export csv_row, csv_header

function csv_escape(field::String)
    return string('"', replace(field, "\"" => "\"\""), '"')
end

function csv_row(fields::Vector{String})
    return join([csv_escape(f) for f in fields], ",")
end

function csv_header()
    return csv_row(["cipher", "rounds", "bytes", "micros"])
end

end

File: fixtures/scale/r/large/core/csv.r
# Minimal CSV encoding for the results file.

csv_escape <- function(field) {
  paste0('"', gsub('"', '""', field), '"')
}

csv_row <- function(fields) {
  paste(vapply(fields, csv_escape, character(1)), collapse = ",")
}

csv_header <- function() {
  csv_row(c("cipher", "rounds", "bytes", "micros"))
}
